--- CiaoDE-1.14.2-13646/makedir/ciaosetup_modules/group_config.sh.orig	2011-07-08 11:46:00.000000000 +0200
+++ CiaoDE-1.14.2-13646/makedir/ciaosetup_modules/group_config.sh	2012-04-08 10:03:01.277155322 +0200
@@ -17,7 +17,7 @@
 EOF
 register_command "configure"
 do__configure() {
-    run_config $*
+    run_config "$@"
 }
 register_command "config_prepare"
 do__config_prepare() {
@@ -26,7 +26,7 @@
 }
 register_command "user_config"
 do__user_config() {
-    run_config --silent=true --instype=src --sysavail=user $*
+    run_config --silent=true --instype=src --sysavail=user "$@"
 }
 
 register_command "menuconfig"
@@ -50,13 +50,13 @@
 #ABSSRC=${CIAODESRC}/${CIAOSRC}
 RUNTIME_CHECKS=$RUNTIME_CHECKS
 UNUSED_PRED_WARNINGS=$UNUSED_PRED_WARNINGS
-CFLAGS=${EXTRA_CFLAGS}
-LDFLAGS=${EXTRA_LDFLAGS}
+CFLAGS="${EXTRA_CFLAGS}"
+LDFLAGS="${EXTRA_LDFLAGS}"
 EOF
 }
 
 run_config() {
-    get_config_options $*
+    get_config_options "$@"
 
     if [ x$HELP = x1 ] ; then
 	config_help
@@ -111,11 +111,11 @@
 		HELP=1
 	    elif expr $ARG : '\(--[^=][^=]*=..*\)' >/dev/null  ; then 
 		OPT=`expr $ARG : '--\([^=][^=]*\)=..*'|sed -e s:_:-:g`
-		VAL=`expr $ARG : '--[^=][^=]*=\(..*\)'`
+		VAL="`expr $ARG : '--[^=][^=]*=\(..*\)'`"
                 if [ x$OPT = xcflags ] ; then
-		    EXTRA_CFLAGS=$VAL
+		    EXTRA_CFLAGS="$VAL"
 		elif [ x$OPT = xldflags ] ; then
-		    EXTRA_LDFLAGS=$VAL
+		    EXTRA_LDFLAGS="$VAL"
 	    # MAKEARG is undocumented because it is only used for debugging.
 		elif [ x$OPT = xmakearg ] ; then
 		    MAKEARG=$VAL
--- CiaoDE-1.14.2-13646/makedir/ciaosetup_modules/command_registry.sh.orig	2011-07-08 11:46:00.000000000 +0200
+++ CiaoDE-1.14.2-13646/makedir/ciaosetup_modules/command_registry.sh	2012-04-08 11:12:44.777291111 +0200
@@ -97,9 +97,9 @@
 	# For compatibility, register two versions of the command name,
 	# one with '_' other with '-'. Both do the same thing.
 	cmd2="`echo "$cmd" | sed -e 's/_/-/g'`" # replace _ by -
-	echo "        ${cmd}) do__${cmd} "'$@'" ;;"
+	echo "        ${cmd}) do__${cmd} \""'$@'"\" ;;"
 	if [ "${cmd}" != "${cmd2}" ]; then
-	    echo "        ${cmd2}) do__${cmd} "'$@'" ;;"
+	    echo "        ${cmd2}) do__${cmd} \""'$@'"\" ;;"
 	fi
     done
     echo "        "'*'") do__unknown "'$act'" ;;"
